client/src/components/Metric.tsx

```diff
@@ -19,6 +20,13 @@ const useStyles = makeStyles({
 
 export const Metric: FunctionComponent = () => {
     const classes = useStyles();
+    const findAllMetrics = useFindAllMetricsQuery();
+    if (findAllMetrics.loading || !findAllMetrics.data || !findAllMetrics.data.findAllMetrics) {
```

Comment:
Can you explain to me what these conditions are checking for?

Reply:
If its loading, or there is no data yet, a loading message is displayed. Theoretically I should only need a loading check but Typescript complains that data can be null and data.findAllMetrics can be null, so I need to add a guard for that too. The data object contains all the metrics once the data has been fetched from the endpoint.

Reply:
I should probably put the error if check first, so that any errors are caught before the loading message is displayed.

Reply:
Fixed. You might get an error if the spring boot server is not running.